app/master/build.py

```diff
@@ -112,10 +112,13 @@ def generate_project_type(self):
         project_type_params = self.build_request.build_parameters()
         project_type_params.update({'build_project_directory': build_specific_project_directory})
         self._project_type = util.create_project_type(project_type_params)
-
         if self._project_type is None:
             raise BuildProjectError('Build failed due to an invalid project type.')
 
+        # Create the build results directory.
+        if not os.path.exists(self._build_results_dir()):
+            os.mkdir(self._build_results_dir())
```

Comment:
I added this here because, in the scenario when there are no atoms and we want to immediately perform the aynchronoous post-build actions, we need to first tar-up the build results directory. But because the build results directory is usually only generated as we receive subjob results from slaves, that errors out.

I figured this is the place where it makes the most sense to generate the initial empty build results directory (generate_project_type).